### PR TITLE
Add build configurations for limited macOS agent builds

### PR DESCRIPTION
**Pull request by big-guy**
### Contributor Checklist
- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/.github/CONTRIBUTING.md)
- [ ] [Sign Gradle CLA](http://gradle.org/contributor-license-agreement/)
- [ ] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: .teamcityTest/Gradle_Check_Tests/CIConfigIntegrationTests.kt [hunk 1]
```diff
@@ -17,6 +18,21 @@ class CIConfigIntegrationTests {
     }
 
     @Test
+    fun macOSBuildsSubset() {
+        val m = CIBuildModel()
+        val p = RootProject(m)
+        val releaseAccept = p.subProjects.find { it.name.contains("Release Accept") }
+        val macOS = releaseAccept?.subProjects?.find { it.name.contains("Macos") }
```
Comment (wolfs) [inline .teamcityTest/Gradle_Check_Tests/CIConfigIntegrationTests.kt]: Please make `releaseAccept` and `macOS` non-null, by appending `!!` to the expressions. Then we can get rid of all the `?`.

File: .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt [hunk 2]
```diff
@@ -20,6 +20,7 @@ val gradleParameters: List<String> = asList(
         java7HomeLinux
 )
 
+// TODO: Does this work on macOS?
```
[the remark this answers is not in the document]
Comment (big-guy) [reply .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt]: done

File: .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt [hunk 3]
```diff
@@ -84,10 +85,16 @@ fun ProjectFeatures.buildReportTab(title: String, startPage: String) {
...
-fun applyDefaults(model: CIBuildModel, buildType: BaseGradleBuildType, gradleTasks: String, notQuick: Boolean = false, runsOnWindows: Boolean = false, extraParameters: String = "", timeout: Int = 90, extraSteps: BuildSteps.() -> Unit = {}) {
-    applyDefaultSettings(buildType, runsOnWindows, timeout)
+fun applyDefaults(model: CIBuildModel, buildType: BaseGradleBuildType, gradleTasks: String, notQuick: Boolean = false, os: OS = OS.linux, extraParameters: String = "", timeout: Int = 90, extraSteps: BuildSteps.() -> Unit = {}) {
+    applyDefaultSettings(buildType, os, timeout)
+
+    val java7HomeParameter = when (os) {
+        OS.windows -> java7Windows
+        OS.linux -> java7HomeLinux
+        // We only have Java 8 on macOS
+        OS.macos -> "-Djava7.home=%macos.java8.oracle.64bit%"
```
[the remark this answers is not in the document]
Comment (big-guy) [reply .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt]: done

File: .teamcityTest/Gradle_Check_Tests/CIConfigIntegrationTests.kt (hunk 1 above)
Comment (big-guy) [reply .teamcityTest/Gradle_Check_Tests/CIConfigIntegrationTests.kt]: done

**Comment by big-guy** [comment]
@wolfs could you give this another look?

**Pull request merged**